src/pages/iou/request/step/IOURequestStepParticipants.js

```diff
@@ -51,8 +55,22 @@ function IOURequestStepParticipants({
         IOUUtils.navigateToStartStepIfScanFileCannotBeRead(receiptFilename, receiptPath, () => {}, iouRequestType, iouType, transactionID, reportID);
     }, [receiptPath, receiptFilename, iouRequestType, iouType, transactionID, reportID]);
 
+    useEffect(() => {
+        if (!newIouType.current) {
+            return;
+        }
+        IOU.updateMoneyRequestTypeParams(routes, newIouType.current);
+        newIouType.current = null;
+    }, [routes, participants]);
```

Comment:
Nope. This is to help reduce the lag where the participant Onyx is late to update, so I want to make sure the participant is updated first. But somehow, this morning I retested it (updating both params and participant at the same time), and the flow works really smoothly and fast.

I don't know exactly what happened with the performance when I worked on the PR, but I think we can keep this.

Reply:
```
// Participants can be added as normal or split participants. We want to wait for the participants' data to be updated before
// updating the money request type route params reducing the amount of work at the same time.
```

Do you think this is enough?

Reply:
`to avoid the overhead when updating both at the same time.`

Sounds good?

Reply:
Updated

Reply:
@parasharrajat So, I found one issue with this.
1. Open money request participants page
2. Press user A to make a manual request. (participant: [A])
3. Go back to the participant page. (participant: [A])
4. Press the "Split" button on the same user. (participant: [A], because the array content is the same, the Onyx is not updated)
5. Nothing happens

and it's because the above `useEffect` won't be triggered.

We can say that this logic that we removed can prevent the above issue, but this is only called when going back and not when closing the app, so it's not 100% reliable.
https://github.com/Expensify/App/blob/c31b51bb72bea65cb81b3fbafe81ff231693a38d/src/pages/iou/request/step/IOURequestStepConfirmation.js#L107-L111

I'm thinking of making `setMoneyRequestParticipants_temporaryForRefactor` a promise by returning the `Onyx.merge` inside it, so we can update the route params inside the promise.
```
IOU.setMoneyRequestParticipants_temporaryForRefactor(transactionID, val).then(() => {
    if (newIouType) {
        IOU.updateMoneyRequestTypeParams(routes, newIouType);
    }
});
```

Let me know what you think!

Reply:
@parasharrajat don't forget to check this one.

Reply:
> Has it been done anywhere in the App?

Let me search for it.

Reply:
I can only find one that returns the onyx and wait for the promise inside the component
https://github.com/Expensify/App/blob/702303c537abbd8721bc90cd05f25558b5922759/src/libs/actions/Transaction.ts#L244-L246
https://github.com/Expensify/App/blob/702303c537abbd8721bc90cd05f25558b5922759/src/pages/iou/request/step/IOURequestStepDistance.js#L156-L158

But there are some that returns the onyx and use it inside a non-component
https://github.com/Expensify/App/blob/702303c537abbd8721bc90cd05f25558b5922759/src/libs/actions/BankAccounts.ts#L36-L44
https://github.com/Expensify/App/blob/702303c537abbd8721bc90cd05f25558b5922759/src/libs/actions/ActiveClients.ts#L4-L6

Reply:
> Before we make this change we should propose this on Slack.

Hmm, I think it's unnecessary as it's not a new thing. Btw, I did this approach in this ongoing [PR](https://www.github.com/Expensify/App/pull/34124) too. I think it's 100% fine if we need to do something sequentially.

I'm asking the confirmation here just so you are aware. As you are fine with it, I have pushed the changes. Please check 🙇

Reply:
I didn't use the Problem/Solution approach as I think the format doesn't suit our case.

Here is the link: https://expensify.slack.com/archives/C01GTK53T8Q/p1704818739022579

Reply:
@parasharrajat looks like we got no answer from the internal team yet, but let's say we are not allowed to do it, I have another alternative.

We will use the `useEffect` back. Then, how do we fix the case where the selected participants are the same? We can update the params immediately knowing the participants' onyx won't be updated.

<img width="729" alt="image" src="https://github.com/Expensify/App/assets/50919443/751f1d98-c197-4ab1-93e3-34d885be3374">

What do you think?

Reply:
@parasharrajat So, the conclusion is we are not allowed to do the promise chaining. This left us with the alternate option above

Reply:
Done